stan/math/prim/mat/err/check_vector.hpp

```diff
@@ -42,6 +43,19 @@ inline void check_vector(const char* function, const char* name,
                    msg_str.c_str());
 }
 
+template <typename T>
+inline void check_vector(const char* function, const char* name, T& x) {
+  if (x.rows() == 1 || x.cols() == 1)
+    return;
+
+  std::ostringstream msg;
+  msg << ") has " << x.rows() << " rows and " << x.cols()
+      << " columns but it should be a vector so it should "
+      << "either have 1 row or 1 column";
```

Comment:
I actually removed this function / added a method for matrix_cl's. If we want to change the wording for this function's error should we do it in both check_vector for eigen matrices and matrix_cls ?

Reply:
tl;dr is that this template was wayyy to general and was so that type deduction would work for matrix_cl's. So I removed this function and made a `matrix_cl` specific version in the `opencl` folder